proto/ibc/core/commitment/v1/commitment.proto

```diff
@@ -26,7 +26,7 @@ message MerklePrefix {
 // arbitrary structured object (defined by a commitment type).
 // MerklePath is represented from root-to-leaf
 message MerklePath {
-  repeated string key_path = 1;
+  repeated bytes key_path = 1;
```

Comment:
This is a proto breaking change, what's the tradeoff here vs. field deprecation?

Reply:
Hey @erwanor! Does this raise any concern in particular for you and do you depend on the message type? 

This is a bit of a protobuf grey area. Myself and @colin-axner looked into this quite a bit to ensure it was okay to make this change in order to support proving kvs stored under non utf8 encoded keys (as requested by union).

Thus far we've considered this a go api breaking but not actually proto encoding breaking. Backwards compatibility should be maintained with this change. I wrote a quick test to validate this, introducing a temp `LegacyMerklePath` type with `repeated string` (old diff).

```go
func (suite *MerkleTestSuite) TestMerklePathEncoding() {
	encodingCfg := moduletestutil.MakeTestEncodingConfig()

	keyPath := []string{"ibc", "commitments/ports/transfer/channels/channel-0/sequences/1"}
	merklePath := types.NewMerklePath(keyPath...)

	bz, err := encodingCfg.Codec.Marshal(&merklePath)
	suite.Require().NoError(err)

	var legacy types.LegacyMerklePath
	err = encodingCfg.Codec.Unmarshal(bz, &legacy)
	suite.Require().NoError(err)

	suite.Require().Equal(keyPath, legacy.KeyPath)
}
```

Some of the resources which helped us come to this conclusion:

https://protobuf.dev/programming-guides/encoding/#structure
> The binary version of a message just uses the field’s number as the key – the name and declared type for each field can only be determined on the decoding end by referencing the message type’s definition (i.e. the .proto file).

https://protobuf.dev/programming-guides/proto2/#updating
> `string` and `bytes` are compatible as long as the bytes are valid UTF-8.

It's possible that we may have overlooked something wrt to rust users/tooling, please let us know if you have any concerns!

Reply:
The biggest concern for me here was that when encoding to JSON you now end up with a `key_path` field which is a base64 encoded byte string.

Reply:
Also happy to introduce a `v2` proto pkg and adjust ibc-go code as necessary. If it makes things easier to keep the old type around!